service/matching/db.go

```diff
@@ -51,16 +51,17 @@ const (
 type (
 	taskQueueDB struct {
 		sync.Mutex
-		namespaceID     namespace.ID
-		taskQueue       *taskQueueID
-		taskQueueKind   enumspb.TaskQueueKind
-		rangeID         int64
-		ackLevel        int64
-		userData        *persistencespb.VersionedTaskQueueUserData
-		userDataChanged chan struct{}
-		store           persistence.TaskManager
-		logger          log.Logger
-		matchingClient  matchingservice.MatchingServiceClient
+		namespaceID      namespace.ID
+		taskQueue        *taskQueueID
+		taskQueueKind    enumspb.TaskQueueKind
+		rangeID          int64
+		ackLevel         int64
+		userData         *persistencespb.VersionedTaskQueueUserData
+		userDataChanged  chan struct{}
+		userDataDisabled error
```

Comment:
How about just use a state enum? Like, the state of user data is that it's disabled temporarily, or it's disabled permanently, or it's enabled. I get that this makes it simpler to just return the exact same field instead of translating the state into an error, but it feels like cutting corners, and it's really unintuitive. If you must keep it this way, I'd rename it to `userDataDisabledReason` at least.

Reply:
Hmm. Unintuitive is subjective of course, but to me it seems like less mental overhead than having a separate set of enums and mapping them into errors. I can do that though, it's simple enough